app/lib/audit/models.dart

```diff
@@ -0,0 +1,129 @@
+  String summary;
+
+  /// List of [User.userId] for the users involved in this record.
+  ///
+  /// This is the list of users who should have this record show up in their personal audit-log.
+  /// This should include [agent], as well as any users who is subject of this action, that's
+  /// it should include the user who did this action, and the user who is invited/removed.
+  ///
+  /// This property **MUST** only be used for querying. The contents of this property is
+  /// **NOT** intended for consumption or presentation by other means. It's only here
+  /// to make it easy to find [AuditLogRecord]s relevant for a specific user.
+  @db.StringListProperty()
+  List<String> users;
+
+  /// List of packages involved in this record.
+  ///
+  /// This is the list of packages who should have this record show up in their
+  /// audit-log.
+  ///
+  /// This property **MUST** only be used for querying. The contents of this property is
+  /// **NOT** intended for consumption or presentation by other means. It's only here
+  /// to make it easy to find [AuditLogRecord]s relevant for a specific package.
+  @db.StringListProperty()
+  List<String> packages;
+
+  /// List of package versions involved in this record.
+  ///
+  /// This is the list of package version who should have this record show up in
+  /// their audit-log.
+  ///
+  /// This property **MUST** only be used for querying. The contents of this property is
+  /// **NOT** intended for consumption or presentation by other means. It's only here
+  /// to make it easy to find [AuditLogRecord]s relevant for a specific package version.
+  @db.StringListProperty()
+  List<String> packageVersions;
+
+  /// List of publishers involved in this record.
+  ///
+  /// This is the list of publishers who should have this record show up in their
+  /// audit-log.
+  ///
+  /// This property **MUST** only be used for querying. The contents of this property is
+  /// **NOT** intended for consumption or presentation by other means. It's only here
+  /// to make it easy to find [AuditLogRecord]s relevant for a specific publisher.
+  @db.StringListProperty()
+  List<String> publishers;
+
+  AuditLogRecord();
+
+  factory AuditLogRecord.packagePublished({
+    @required User uploader,
+    @required String package,
+    @required String version,
+    @required DateTime created,
+  }) {
+    final summary = 'Package `$package` '
+        'version `$version` '
+        'was published '
+        'by `${uploader.email}`.';
+    return AuditLogRecord()
+      ..id = createUuid()
+      ..created = created
+      ..expires = _expiresInFarFuture
+      ..kind = AuditLogRecordKind.packagePublished
+      ..agent = uploader.userId
+      ..summary = summary
+      ..users = [uploader.userId]
+      ..packages = [package]
+      ..packageVersions = ['$package/$version']
+      ..publishers = [];
+  }
+}
+
+abstract class AuditLogRecordKind {
+  static const packagePublished = 'package-published';
```

Comment:
```suggestion
  /// Event that a package was published.
  ///
  /// This can be an entirely new package or just a new version to an existing package.
  static const packagePublished = 'package-published';
```